src/mysql_k8s_helpers.py

```diff
@@ -852,3 +852,7 @@ def set_cluster_primary(self, new_primary_address: str) -> None:
         """Set the cluster primary and update pod labels."""
         super().set_cluster_primary(new_primary_address)
         self.update_endpoints()
+
+    def fetch_error_log(self) -> Optional[str]:
+        """Fetch the MySQL error log."""
+        return self.read_file_content("/var/log/mysql/error.log")
```

Comment:
can this be a bad idea? if the error log is large, it will be loaded from disk into memory

instead, may be consider iterating through lines of the file and searching for the error code you are looking for

Reply:
I considered that, but given that we are rotating it (and inhibiting the rotation solely to the duration of the upgrade), it should not be enormous (as in MB)